pkg/client/record/event.go

```diff
@@ -145,7 +145,11 @@ func recordEvent(sink EventSink, event *api.Event, updateExistingEvent bool) boo
 	var err error
 	if updateExistingEvent {
 		newEvent, err = sink.Update(event)
-	} else {
+	}
+	// Update can fail because the event may have been removed and it no longer exists.
+	if !updateExistingEvent || (updateExistingEvent && err != nil) {
+		// Making sure that ResourceVersion is empty on creation
+		event.ResourceVersion = ""
```

Comment:
In addition to resetting the resourceVersion, since a new event is being created, we should reset the event object:
  * Set `event.count` back to 1
  * Set `event.FirstTimestamp` to `event.LastTimestamp`